kolibri/plugins/learn/assets/src/actions.js

```diff
@@ -151,117 +156,109 @@ function redirectToExploreChannel(store) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', PageNames.EXPLORE_ROOT);
 
-  _getCurrentChannelId()
-    .then((currentChannelId) => {
-      store.dispatch('CORE_SET_ERROR', null);
-      if (currentChannelId) {
-        store.dispatch('SET_CURRENT_CHANNEL', currentChannelId);
-        cookiejs.set('currentChannel', currentChannelId);
+  ChannelResource.getCollection({}).fetch().then(
+    (channelsData) => {
+      const channelList = _channelListState(channelsData);
+      const channelId = _getDefaultChannelId(store, channelList);
+      _handleChannels(store, channelId, channelList);
+      if (channelList.length) {
         router.replace({
           name: constants.PageNames.EXPLORE_CHANNEL,
-          params: {
-            channel_id: currentChannelId,
-          },
+          params: { channel_id: channelId },
         });
       } else {
         router.replace({ name: constants.PageNames.CONTENT_UNAVAILABLE });
       }
-    })
-    .catch((error) => {
+    },
+    (error) => {
       store.dispatch('CORE_SET_ERROR', JSON.stringify(error, null, '\t'));
       store.dispatch('CORE_SET_PAGE_LOADING', false);
-    });
+      document.title = _errorTitle();
+    }
+  );
 }
 
+
 function redirectToLearnChannel(store) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', PageNames.LEARN_ROOT);
 
-  _getCurrentChannelId()
-    .then((currentChannelId) => {
-      store.dispatch('CORE_SET_ERROR', null);
-      if (currentChannelId) {
-        store.dispatch('SET_CURRENT_CHANNEL', currentChannelId);
-        cookiejs.set('currentChannel', currentChannelId);
+  ChannelResource.getCollection({}).fetch().then(
+    (channelsData) => {
+      const channelList = _channelListState(channelsData);
+      const channelId = _getDefaultChannelId(store, channelList);
+      _handleChannels(store, channelId, channelList);
```

Comment:
As above with explore. Is there a way we could avoid repeating this code between the two, also?